agent/consul/server.go

```diff
@@ -378,6 +382,9 @@ func NewServer(config *Config, flat Deps, publicGRPCServer *grpc.Server) (*Serve
 	loggers := newLoggerStore(serverLogger)
 
 	recorder := middleware.NewRequestRecorder(serverLogger)
+
+	eventPublisher := stream.NewEventPublisher(10 * time.Second)
```

Comment:
If we need an event publisher outside of the Server in the future then we could move this further out into the Deps struct.